### PR TITLE
This fixes test_modulo_complex in test_float.java

### PR DESCRIPTION
**Pull request by student-06**
#36 this operation float%complex now gives appropriate error.

### CODE REVIEW

File: python/common/org/python/types/Float.java [hunk 1]
```diff
@@ -355,7 +355,9 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
             } else {
                 throw new org.python.exceptions.ZeroDivisionError("float division by zero");
             }
-        }
+        }// else if (other instanceof org.python.types.Complex){
+           // return other.__truediv__(this);
+        //}
```
Comment (eliasdorneles) [inline python/common/org/python/types/Float.java]: Please don't add commented out code, these don't add anything useful and only confuse someone reading the code.

File: python/common/org/python/types/Float.java [hunk 2]
```diff
@@ -423,9 +425,11 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     double result = (((((double) this.value) % other_val) + other_val) % other_val);
                     return new org.python.types.Float(result);
                 }
+            } else if (other instanceof org.python.types.Complex){
+                throw new org.python.exceptions.TypeError("can't mod complex numbers.");
             }
         } catch (org.python.exceptions.TypeError e) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for %: 'float' and '" + other.typeName() + "'");
+            throw e;
```
Comment (eliasdorneles) [inline python/common/org/python/types/Float.java]: If we're just rethrowing this exception, maybe we should simply not catch `TypeError` and let it raise.

File: python/common/org/python/types/Float.java [hunk 3]
```diff
@@ -456,7 +460,7 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
         }
 
         if (other instanceof org.python.types.Int) {
-            long other_val = ((org.python.types.Int) other).value;
+            long other_val = ((org.python.types.Int) other).value; // exponent
```
Comment (eliasdorneles) [inline python/common/org/python/types/Float.java]: Was this change intended?
Seems a bit out of scope for this PR. Anyway, instead of adding a comment, a more useful change would be to rename the variable to be `exponent`.

File: python/common/org/python/types/Float.java (hunk 1 above)
Comment (student-06) [reply python/common/org/python/types/Float.java]: Oh sorry for this. I have removed comments. I won't repeat this again and thanks.

File: python/common/org/python/types/Float.java (hunk 3 above)
Comment (student-06) [reply python/common/org/python/types/Float.java]: Oh sorry my bad. I was just reading code and wrote some comments over here and there. I will remove those. Thanks again.

File: python/common/org/python/types/Float.java (hunk 2 above)
Comment (student-06) [reply python/common/org/python/types/Float.java]: So should I remove try catch block ?

**Comment by eliasdorneles** [comment]
Looks good, thanks @student-06 !
